misc/bump/README.md

```diff
@@ -0,0 +1,138 @@
+patterns (combinations of array access with either constant index offsets or
+struct fields). In those cases, it might be worth obtaining an opinion from
+someone with `SimplExportAndRefine` experience. Otherwise, don't worry about it
+until it fails.
+
+If you think no proof update is required, approve the seL4 pull request to be
+merged. As soon as possible after it is merged, follow the bump script recipe
+below. If you think a proof update is required, or if you bump and regression
+subsequently fails, then flag the need for a verification patch.
+
+## Verification patches
+
+A proof update for an seL4 update requires merging pull requests for both `l4v`
+and `seL4`, and a manifest bump. The proof engineer who prepared the patch
+should already have made sure the seL4 pull request is rebased and tested the
+proof update against the rebased version of the seL4.
+
+It doesn't hurt to double check that the manifest displayed in the proof test
+log contains the commit hashes of the branches you want to merge, but if you are
+sure that everything is up to date, you can skip this step.
+
+Once you're happy that both PRs have been tested and reviewed appropriately,
+both can be merged. The seL4 PR should be merged first, followed without too
+much delay (see below) by the l4v PR. As soon as both PRs are merged, follow the
+bump script recipe below.
+
+Timing: in between the three merges, the repos are in an inconsistent state with
+respect to each other. Don't worry if the merges take too long (Bamboo checks
+roughly every 5 min), the only adverse effect is a wasted test cycle that tests
+the seL4 PR without the l4v PR and/or a failure of the preprocess test. But do
+try to minimise the time the repositories are in an inconstent state with
+respect to each other, so that others can keep working.
+
+## Using the bump script
+
+The [bump script][] can be found at `l4v/misc/bump/bump-ver-manifest` in any
+checkout of [l4v][]. Unless you know `repo` well, it is advisable to keep a
+separate `repo` checkout of [verification-manifest][] that you only use for
+manifest bumps.
+
+[bump script]: https://github.com/seL4/l4v/tree/master/misc/bump/
+
+To bump the manifest:
+
+1. `repo sync` your [verification-manifest][] checkout.
+
+2. Ensure that for `l4v`, `HEAD` corresponds to current master, and that this is
+   the `l4v` commit that verifies the new `seL4` version.
+
+3. Note that for `seL4`, new commits will have been fetched from the remote, but
+   `HEAD` will still correspond to the `seL4` commit in the *current* manifest.
+
+4. In the `seL4` directory, look at `git log ^HEAD verification/master`. Check
+   that it shows exactly the commits in the `seL4` PR for which the bump is
+   required.
+
+5. In the `seL4` directory, perform `git checkout verification/master`.
+
+6. In the root of the `repo` checkout (i.e. parent of `l4v` directory), run
+   `./l4v/misc/bump/bump-ver-manifest` and follow the prompts.
```

Comment:
When I originally wrote this, I think I unintentionally made the assumption that before performing these steps, the `seL4` directory is checked out to `master`, i.e. not a detached HEAD. I'm not sure if step 4 works if you start with a detached HEAD. Then again, I think the script tells you what commits will be incorporated, so perhaps step 4 is not necessary.

Reply:
Actually, it might work just fine.

Reply:
Yeah, I was imagining that step 4 was `git log ^master verification/master`, which is what I habitually do, because I also habitually have seL4 checked out to `master`. So I think this is fine.

Reply:
Cool, will leave as is, then. I also usually have it at `master` so didn't spot it.